app/Http/Controllers/Bible/LibraryController.php

```diff
@@ -451,14 +451,20 @@ public function volume()
         $filesets = cacheRemember('v2_library_volume', $cache_params, now()->addDay(), function () use ($dam_id, $media, $language_name, $iso, $updated, $organization, $version_code) {
             $language_id = $iso ? optional(Language::where('iso', $iso)->first())->id : null;
 
+            if (!empty($iso) && empty($language_id)) {
```

Comment:
I had not considered the case for the volume action where the language_id parameter is a value similar to "ZZZZ" so, it is not a language but the query was returning all records similar when there not filters.